src/backend/src/lib.rs

```diff
@@ -511,9 +515,33 @@ fn list_custom_tokens() -> Vec<CustomToken> {
 
 #[update(guard = "caller_is_not_anonymous")]
 #[allow(clippy::needless_pass_by_value)]
-#[allow(unused_variables)]
-fn add_user_credential(request: AddUserCredentialRequest) {
-    // TODO: Implement https://dfinity.atlassian.net/browse/GIX-2649
+fn add_user_credential(request: AddUserCredentialRequest) -> Result<(), AddUserCredentialError> {
+    let user_principal = ic_cdk::caller();
+    let stored_principal = StoredPrincipal(user_principal);
+    let current_time_ns = time() as u128;
```

Comment:
I would cast the time where the cast is required, assuming it's for a specific usage. This for readability reasons.

Reply:
Done!